### PR TITLE
Fix nav menu border in High Contrast

### LINKED ISSUE #2774: Fix nav menu border in High Contrast in FF
**Describe the bug**
There have been a lot of changes to the menu button lately and at some point High Contrast mode in FireFox stopped working. On hover/focus border and icon are not located correctly.
Screenshot 2019-12-04 at 15 57 50

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to Psammead Storybook
2. Click on Navigation - Canonical - Menu Button
3. See error

**Expected behaviour**
Icon should be centred and border should be located on the edges of a button

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

### PR DESCRIPTION
**Pull request by OlgaLyubin**
Resolves #2774

**Overall change:**
Borders and icon are now properly located on a button and do not get shifted in High Contrast mode on FireFox.
Screenshot 2019-12-05 at 14 37 25

**Code changes:**
The problem before was that (only in FF HC mode) a border appeared which was not supposed to be there. Rearranging the styling fixed the issue.
Screenshot 2019-12-05 at 14 35 51

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: packages/components/psammead-navigation/src/DropdownNavigation/index.jsx [hunk 1]
```diff
@@ -156,6 +155,7 @@ const MenuButton = styled(Button)`
       ${iconBorder};
     }
   }
+  border: 0;
```
Comment (DenisHdz) [inline packages/components/psammead-navigation/src/DropdownNavigation/index.jsx]: Can you put this style after the `background-color: transparent;`?

Comment (OlgaLyubin) [reply packages/components/psammead-navigation/src/DropdownNavigation/index.jsx]: Done

**Review by DenisHdz — APPROVED** [review]
LGTM

border-hc

**Review by tochwill — APPROVED** [review]

**Review by AlistairGempf — APPROVED** [review]
Looks good

**Pull request merged**